app/src/components/Provenance.tsx

```diff
@@ -148,27 +184,84 @@ export const Provenance: React.FC<{tokenId: number}> = ({ tokenId }) => {
-          sale: {
-            seller: event.returnValues.from,
-            price: event.returnValues.price,
-            buyer: event.returnValues.to,
-            location: event.returnValues.location,
-            date: event.returnValues.date,
-          },
-        };
-      },
-      ));
-    });
+    const otherRecordTypes = ['Sale', 'Stolen', 'Recovered', 'Damaged', 'Restored', 'Film'];
+    const otherRecords: Promise<ProvenanceRecord[]> = Promise.all(otherRecordTypes
+      .map(async (type: string): Promise<ProvenanceRecord[]> => {
+        const pastEvents = await ArtifactRegistry.getPastEvents('Record' + type, options);
+        const tokenRelevantEvents = pastEvents.filter((event: any) =>
+          event.returnValues.tokenId === tokenId.toString());
+        const resultRecords: ProvenanceRecord[] = [];
+        for (const event of tokenRelevantEvents) {
+          const timestamp = await web3.eth.getBlock(event.blockNumber).then((block) => block.timestamp);
+          const typeLowerCase = type.toLowerCase();
+          const timestampMoment = moment.unix(Number(timestamp));
+          resultRecords.push(typeLowerCase === 'sale'
+            ? {
+              type: typeLowerCase,
+              txDate: timestampMoment,
+              sale: {
+                seller: event.returnValues.from,
+                price: event.returnValues.price,
+                buyer: event.returnValues.to,
+                location: event.returnValues.location,
+                date: event.returnValues.date,
+              },
+            }
+            : {
+              type: typeLowerCase,
+              txDate: timestampMoment,
+              detail: {
+                detailInfo: event.returnValues.detailInfo,
+                date: event.returnValues.date,
+              },
+            });
+        }
+        return resultRecords;
+      }))
+      // list of lists flattened to one big list
+      .then((listOfLists: any) => listOfLists.flat());
+
+    const recordComparator = (a: ProvenanceRecord, b: ProvenanceRecord): number => {
+      if (a.artist && b.artist) {
+        return 0;
+      }
+      if (a.artist && !b.artist) {
+        return -1;
+      }
+      if (!a.artist && b.artist) {
+        return 1;
+      }
+      if (!a.artist && !b.artist) {
+        let aDate: string | null = null;
+        let bDate: string | null = null;
+        if (a.detail) {
+          aDate = a.detail.date;
+        } else if (a.sale) {
+          aDate = a.sale.date;
+        }
+        if (b.detail) {
+          bDate = b.detail.date;
+        } else if (b.sale) {
+          bDate = b.sale.date;
+        }
+        if (!aDate && !bDate) {
+          return 0;
+        }
+        if (aDate && !bDate) {
+          return -1;
+        }
+        if (!aDate && bDate) {
+          return 1;
+        }
+        if (aDate && bDate) {
+          return aDate.localeCompare(bDate);
+        }
+      }
+      return -1;
+    };
```

Comment:
If we are sorting the records surely the first thing we should be inspecting is date not artists?

Reply:
the way the provenance record object works is that records that have artists (aka registration records) are guaranteed to not have dates, and vice versa, so it's unclear where to put such records in the timeline. ive arbitrarily put them at the top

Reply:
okay makes sense!

Reply:
the only record this applies to is the one for when a piece is first registered